### PR TITLE
chore: migrate from kuttl to chainsaw (linodecluster-controller)

### PR DESCRIPTION
**Pull request by eddycharly**
Migrate from kuttl to chainsaw.

### CODE REVIEW

**Comment by eddycharly** [comment]
```
        content: 'URI="nodebalancers" FILTER="{\"label\":\"linodecluster-test-api-server\"}"
          make callLinodeApiGet | grep ''results": 1'''
```
Is this supposed to fail the test if the `grep` does not find the expected results ? 🤔

**Comment by eljohnson92** [comment]
yes, that is the goal of it

**Comment by eddycharly** [comment]
@eljohnson92 but grep does not exit with `1` if not found right ?

**Comment by eddycharly** [comment]
And why do we sometimes have `FILTER="{\"label\":\"linodecluster-test-api-server\"}"` and other times `FILTER="{\"tags\":\"linodecluster-test-api-server\"}"` shouldn't the filter be the same ?

File: e2e-chainsaw/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml
Comment (eddycharly) [inline e2e-chainsaw/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml:42]: @eljohnson92 i replaced the makefile target with an inline script

Comment (eddycharly) [reply e2e-chainsaw/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml:42]: If the payload returned is json it's easy to write a robust assertion

Comment (eljohnson92) [reply e2e-chainsaw/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml:42]: ah, yes, I like the way that is laid out. it is a json response and I would much prefer checking values within the json, is that something we can do natively with Chainsaw or would we have to use JQ to handle checking fields/values?

Comment (eddycharly) [reply e2e-chainsaw/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml]: @eljohnson92 i implemented the robust checks 🤞

Comment (eddycharly) [reply e2e-chainsaw/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml]: @eljohnson92 i implemented the check, it's natively supported

File: e2e-chainsaw/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml
Comment (eddycharly) [inline e2e-chainsaw/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml:30]: Those two vars could be set globally 🤷

**Comment by eddycharly** [comment]
Please allow workflow to run, i'd like to see how it works 🙏

**Comment by eljohnson92** [comment]
@eddycharly we have an issue because we are using API tokens to talk to the Linode API it can't be run by forks, I'm trying to see if there is a quick fix I can get in to allow us to trigger this workflow even from forks

**Comment by eddycharly** [comment]
@eljohnson92 ah i see, this PR doesn't remove kuttl tests, it runs both kuttl and chainsaw side by side.

**Comment by eddycharly** [comment]
Indeed
```
capl-control… │ [manager] 2024-03-04T22:47:48Z	ERROR	setup	unable to start operator	{"error": "failed to get LINODE_TOKEN environment variable"}
```

**Comment by eljohnson92** [comment]
I was able to test this locally so I would be open to merging this, before we do that could we change this to just replace the existing e2e test version?

**Comment by eddycharly** [comment]
> I was able to test this locally so I would be open to merging this, before we do that could we change this to just replace the existing e2e test version of minimal-linodecluster?

You want me to remove the kuttl version of the minimal-linodecluster test ?

**Comment by eddycharly** [comment]
> instead of creating a separate e2e-chainsaw/ directory structure I'd rather we just replace the existing KUTTL test with this new chainsaw test in place. unless that will make test targets way too messy. it would be good to see the diff here in place for review

Doing this is going to be challenging, kuttl will think it can run the test but it will fail.
That's why i created two separate folders, the plan is to replace the kuttl e2e test folder by the chainsaw one at the end (and keep only chainsaw). We will see the diff at this time.

**Comment by eddycharly** [comment]
@eljohnson92 let me try something 🤞

**Comment by eljohnson92** [comment]
understand that might be challenging, I would be fine with merging this separately if it isn't immediately easy to fix

**Comment by eddycharly** [comment]
@eljohnson92 I just pushed, chances are it will work 🤞

**Comment by eljohnson92** [comment]
awesome, just ran another test and it passed. LGTM but I'll let some other folks take a look at it tomorrow before merging it in

**Comment by eddycharly** [comment]
> just ran another test and it passed

Awesome, I will take care of other tests in follow-up PRs.
Thanks!

**Review by AshleyDumaine — APPROVED** [review]

File: e2e/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml
Comment (eddycharly) [reply e2e/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml:45]: Doc is spread between https://kyverno.github.io/chainsaw/latest/operations/check/ and https://kyverno.github.io/kyverno-json/latest/

Chainsaw uses assertion trees from kyverno-json, we didn't repeat the same doc in chainsaw.

Comment (eddycharly) [reply e2e/linodecluster-controller/minimal-linodecluster/chainsaw-test.yaml:45]: I plan to rewrite docs for the v0.2.x series :)

**Pull request merged**
